pkg/metrics/metrics.go

```diff
@@ -199,6 +199,10 @@ var (
 	// endpoints, labeled by span name and status ("success" or "failure")
 	EndpointRegenerationTimeStats = NoOpObserverVec
 
+	// EndpointGCCount is a count of goroutines blocking on attempts to
+	// garbage collect an endpoint from the k8s watcher context.
+	EndpointGCCount = NoOpGauge
```

Comment:
Nit: I think we can be a bit more descriptive on the name to signal that this is an "active" count. Something as simple as `EndpointActiveGCCount` or similar can work. At first glace, `EndpointGCCount` sounds like it's counting how many times we've GC'd endpoints.

Reply:
In the latest iteration, I have renamed this gauge to `cilium_endpoint_active_gc_count`.

I also introduce a new metric `cilium_endpoint_gc_count` which does count the number of times we GC'd endpoints. That way, we can find out if this code is actually effective without having to flood the agent logs.

Reply:
These metrics are now removed.